### PR TITLE
- [cpuinfo] fix M1 build

### PR DESCRIPTION
**Pull request by SSE4**
Specify library name and version:  **cpuinfo/all**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/cpuinfo/all/conanfile.py [hunk 1]
```diff
@@ -72,6 +72,8 @@ def _configure_cmake(self):
         self._cmake.definitions["CLOG_RUNTIME_TYPE"] = "default"
         self._cmake.definitions["CLOG_BUILD_TESTS"] = False
         self._cmake.definitions["CMAKE_POSITION_INDEPENDENT_CODE"] = self.options.get_safe("fPIC", True)
+        if self.settings.arch == "armv8":
+            self._cmake.definitions["CMAKE_SYSTEM_PROCESSOR"] = "armv8"
```
Comment (SpaceIm) [inline recipes/cpuinfo/all/conanfile.py:76]: Is not set by conan? Seems to be a hack.

Comment (SSE4) [reply recipes/cpuinfo/all/conanfile.py:76]: for me, it wasn't set by conan. I doubt CMake helper (especially an old one) will ever set it... see https://github.com/conan-io/conan/pull/8026

Comment (madebr) [inline recipes/cpuinfo/all/conanfile.py:76]: Passing `CMAKE_SYSTEM_PROCESSOR` as argument without a `CMAKE_TOOLCHAIN_FILE`, does this set `CMAKE_CROSSCOMPILING`?

Comment (madebr) [reply recipes/cpuinfo/all/conanfile.py:76]: Thanks for the reply.

I was confusing `CMAKE_SYSTEM_NAME` and `CMAKE_SYSTEM_PROCESSOR`.
Both are supposed to be set by cmake toolchains.
Accidentally enabling `CMAKE_CROSSCOMPILING` might modify the build logic somewhere unwanted.

But we're good here. Thanks!

**Review by madebr — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**